### PR TITLE
refactor(batch): move the storage access of `RowSeqScan` from the executor builder to the body

### LINKED ISSUE #5270: batch: do not create the iterator or point-get the row in `SeqScan` builder
Currently, we create the storage iterator and even point-get the resulting row in the executor builder of `SeqScan`, which looks weird. We expect the building function to be simply "proto to instance" instead of including some I/O.

https://github.com/risingwavelabs/risingwave/blob/6289e8a8add6ef300aa344ec831e80972b1375f0/src/batch/src/executor/row_seq_scan.rs#L246-L263

A possible refactor might be resolving the range in the executor body, or directly storing the `Future` in the `ScanType`.

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR moves the storage access of `RowSeqScan` from the executor builder to the body, so the time measurement of "building executors" can be measured more preciously.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- Close #5270

### CODE REVIEW

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Comment by chenzl25** [comment]
The usage of future make the code not straightforward.  Ideally, I think the structure of `RowSeqScanExecutor` should contain something like `PkBound` so that it can construct an iterator during execution. While `new_boxed_executor()` just convert the proto into `PkBound`.
```rust
/// Executor that scans data from row table
pub struct RowSeqScanExecutor<S: StateStore> {
    chunk_size: usize,
    pk_len: usize,
    identity: String,

    /// Batch metrics.
    /// None: Local mode don't record mertics.
    metrics: Option<BatchTaskMetricsWithTaskLabels>,

    epoch: HummockReadEpoch,
    table: StorageTable<S>,

    scan_ranges: Vec<PkBound>,
}

struct PkBound {
    prefix: Row,
    bound: (Bound<Datum>, Bound<Datum>),
}
```

File: src/batch/src/executor/row_seq_scan.rs [hunk 1]
```diff
@@ -310,57 +303,76 @@ impl<S: StateStore> Executor for RowSeqScanExecutor<S> {
...
-                    }
-                }
+        let ScanRange {
+            pk_prefix,
+            next_col_bounds,
+        } = scan_range;
+
+        // Resolve the scan range to scan type.
+        if pk_prefix.size() == 0 && ScanRange::is_full_range(&next_col_bounds) {
+            unreachable!()
```
Comment (chenzl25) [inline src/batch/src/executor/row_seq_scan.rs:336]: It's a possible path. e.g. `ScanRange::full()` generated before.

**Review by chenzl25 — CHANGES_REQUESTED** [review]

**Review by chenzl25 — APPROVED** [review]
LGTM!

**Pull request merged**